pennylane/devices/default_qubit.py

```diff
@@ -142,6 +180,148 @@ def _apply_operation(self, operation):
         else:
             self._apply_unitary(matrix, wires)
 
+    def _apply_x(self, state, axes, **kwargs):
+        """Applies a PauliX gate by rolling 1 unit along the axis specified in ``axes``.
```

Comment:
Might be worth elaborating on why this is equivalent to applying a PauliX

Reply:
Thanks, I tried to add an explanation, but I have to admit these things are hard to explain!

789a724bbd847a160c633ecf77d872c61c2c9c48